src/integrations/bypass-post-meta-integration.php

```diff
@@ -0,0 +1,161 @@
+	}
+
+	/**
+	 * Filter updating the post meta.
+	 *
+	 * @param null|bool $check      Whether to allow updating metadata for the given type.
+	 * @param int       $object_id  ID of the object metadata is for.
+	 * @param string    $meta_key   Metadata key.
+	 * @param mixed     $meta_value Metadata value. Must be serializable if non-scalar.
+	 * @param mixed     $prev_value Optional. Previous value to check before updating.
+	 *                              If specified, only update existing metadata entries with
+	 *                              this value. Otherwise, update all entries.     *
+	 *
+	 * @return bool
+	 */
+	public function filter_update_user_meta( $check, $object_id, $meta_key, $meta_value, $prev_value ) {
+		return $this->filter_update_meta( $check, $object_id, $meta_key, $meta_value, 'user' );
+	}
+
+	/**
+	 * Filter updating a meta field.
+	 *
+	 * @param null|bool $check      Whether to allow updating metadata for the given type.
+	 * @param int       $object_id  ID of the object metadata is for.
+	 * @param string    $meta_key   Metadata key.
+	 * @param mixed     $meta_value Metadata value. Must be serializable if non-scalar.
+	 * @param mixed     $meta_type  One of 'user', 'post' or 'term'.
+	 *
+	 * @return bool
+	 */
+	private function filter_update_meta( $check, $object_id, $meta_key, $meta_value, $meta_type ) {
+		if ( strpos( $meta_key, '_yoast_wpseo_' ) === 0 ) {
+			$key = str_replace( '_yoast_wpseo_', '', $meta_key );
+			if ( ! isset( $this->map_meta_indexable[ $key ] ) ) {
+				return $check;
+			}
+			$field             = $this->map_meta_indexable[ $key ];
+			$indexable         = $this->indexable_repository->find_by_id_and_type( $object_id, $meta_type, false );
+			$indexable->$field = $meta_value;
+
+			return $indexable->save();
+		}
+
+		return $check;
+	}
+
+	/**
+	 * Filters get_{post|term|user}_meta to retrieve the value from Indexables instead.
+	 *
+	 * @param mixed  $value     The value to return, either a single metadata value or an array
+	 *                          of values depending on the value of `$single`. Default null.
+	 * @param int    $object_id ID of the object metadata is for.
+	 * @param string $meta_key  Metadata key.
+	 * @param bool   $single    Whether to return only the first value of the specified `$meta_key`.
+	 * @param string $meta_type Type of object metadata is for. Accepts 'post', 'comment', 'term', 'user',
+	 *                          or any other object type with an associated meta table.
+	 *
+	 * @return mixed
+	 */
+	public function filter_get_meta( $value, $object_id, $meta_key, $single, $meta_type = 'post' ) {
+		if ( strpos( $meta_key, '_yoast_wpseo_' ) === 0 ) {
+			$key = str_replace( '_yoast_wpseo_', '', $meta_key );
+			if ( ! isset( $this->map_meta_indexable[ $key ] ) ) {
+				return $value;
+			}
+			$field     = $this->map_meta_indexable[ $key ];
+			$indexable = $this->indexable_repository->find_by_id_and_type( $object_id, $meta_type, false );
```

Comment:
This could also return `false` if the indexable hasn't been created yet ( not creating it is fine as we there'd be no data to read.

If the `$indexable` is `false` we should likely return `false` here as well.

Reply:
we should return the `$check` value, but yeah agreed. fixing.